packages/react/src/components/InAppMessaging/MessageLayout/utils.ts

```diff
@@ -0,0 +1,21 @@
+import tinycolor from 'tinycolor2';
+
+import { MessageComponentStyles } from '../hooks/useMessageProps';
+
+import { MessageLayoutButtonModifier } from './types';
+
+/**
+ * Utility for determining the class modifier for an In-App Messaging button
+ * based on its background color
+ *
+ * @param buttonStyles button styles which should contain the background color
+ *
+ * @returns the modifier - either 'light' or 'dark'
+ */
+export const getButtonModifier = (
+  buttonStyles: MessageComponentStyles['primaryButton' | 'primaryButton']
+): MessageLayoutButtonModifier => {
+  const { backgroundColor } = buttonStyles ?? {};
+  const color = tinycolor(backgroundColor);
+  return color.isDark() ? 'dark' : 'light';
```

Comment:
Could you expand a bit on why we need to derive dark/light from bg color instead of relying on `Themeprovider` 's colorMode?

Reply:
We receive the `backgroundColor` property from the `InAppMessage` payload, which does not currently support variants. Because the user is able to set the `backgroundColor` in Pinpoint, this utility allows us to know whether or not the payload `backgroundColor` translates to a light or dark color

Reply:
I see, thanks, good to know. Would be nice to include this info in a comment for future reference. :)